src/main/py/peer/peer.py

```diff
@@ -178,11 +180,14 @@ async def announce_trade(self, wait_time=timedelta(seconds=10)):
         offer_announcement = OfferAnnouncement(artwork)
         announcement_key = utils.generate_random_sha1_hash()
         self.inventory.add_pending_trade(announcement_key, offer_announcement)
+
         asyncio.get_event_loop().call_later(
-            wait_time,
-            asyncio.create_task,
-            self.handle_announcement_deadline(announcement_key, offer_announcement),
+            wait_time.total_seconds(),
```

Comment:
You had mentioned that you had looked up some things related to `asyncio`, so I'm assuming the changes to the `call_later()` call here are related. Documenting the benefits of this change vs the original with any useful reference links would be a great addition to the final commit message to aid understanding.